route-rs-runtime/src/packet/ipv6.rs

```diff
@@ -0,0 +1,336 @@
+use crate::packet::*;
+use std::borrow::Cow;
+
+#[allow(dead_code)]
+pub struct Ipv6Packet<'packet> {
+    pub data: PacketData<'packet>,
+    // There may be various "Extension Headers", so we should figure out the actual offset and store it here for
+    // easy access in the helper functions.
+    pub packet_offset: usize,
+    pub payload_offset: usize,
+}
+
+impl<'packet> Ipv6Packet<'packet> {
+    fn new(packet: PacketData) -> Result<Ipv6Packet, &'static str> {
```

Comment:
Forgot to switch this so the new function takes the variable packet_offset length. Will get that tomorrow.

Reply:
See ipv4::new() for the idea.